src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.trim().length === 0) {
+    return {};
+  }
+
+  const result = {};
+  const pairs = sourceString.split(';');
+
+  pairs.forEach((pair) => {
+    const [key, value] = pair.split(':').map((item) => item.trim());
+
+    if (key && value) {
+      result[key] = value;
+    }
```

Comment:
The current condition checks if both `key` and `value` are truthy. However, this will include cases where `key` or `value` might be an empty string, which may not be desired. Consider adding a check to ensure `key` and `value` are not empty strings to prevent unintended entries in the result object.